drivers/usb_c/ppc/usbc_ppc_numaker.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_ppc
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/usb_c/usbc_ppc.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_numaker.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ppc_numaker, CONFIG_USBC_LOG_LEVEL);
+
+#include <soc.h>
+#include <NuMicro.h>
+
+/* Implementation notes on NuMaker TCPC/PPC/VBUS
+ *
+ * PPC and VBUS rely on TCPC/UTCPD and are just pseudo. They are completely
+ * implemented in TCPC/UTCPD.
+ */
+
+/* Immutable device context */
+struct numaker_ppc_config {
+	const struct device *tcpc_dev;
+};
+
+/* Mutable device context */
+struct numaker_ppc_data {
+	uint32_t dummy;
+};
+
+/* TCPC exported */
+int numaker_tcpc_ppc_is_dead_battery_mode(const struct device *dev);
```

Comment:
These prototypes should be commented and placed in a header file.

Reply:
They are moved to one internal header file.